release/models/bgp/openconfig-bgp-policy.yang

```diff
@@ -1290,6 +1319,184 @@ module openconfig-bgp-policy {
+        description
+          "Active only when the set-as-path method is INLINE";
+      }
+      description
+        "Set the AS path values for the action inline with
+        a list.";
+
+      container config {
+        description
+          "Configuration data for inline specification of
+          set-as-path action";
+
+        uses set-as-path-inline-config;
+      }
+
+      container state {
+
+        config false;
+
+        description
+          "Operational state data or inline specification of
+          set-as-path action";
+
+        uses set-as-path-inline-config;
+        uses set-as-path-inline-state;
+      }
+    }
+  }
+
+  grouping set-as-path-reference-config {
+    description
+      "Configuration data for referening an as-path-set
+      in the set-as-path action";
+
+    leaf-list as-path-set-refs {
+      type leafref {
+        path "/oc-rpol:routing-policy/oc-rpol:defined-sets/" +
+          "oc-bgp-pol:bgp-defined-sets/" +
+          "oc-bgp-pol:as-path-sets/" +
+          "oc-bgp-pol:as-path-set/" +
+          "oc-bgp-pol:as-path-set-name";
+      }
+      description
+        "References a list of defined AS path sets by name";
+    }
+
+    // TODO(xqm): do we need to keep this deperecated leaf?
+    leaf as-path-set-ref {
+      status deprecated;
+      type leafref {
+        path "/oc-rpol:routing-policy/oc-rpol:defined-sets/" +
+          "oc-bgp-pol:bgp-defined-sets/" +
+          "oc-bgp-pol:as-path-sets/" +
+          "oc-bgp-pol:as-path-set/" +
+          "oc-bgp-pol:as-path-set-name";
+      }
+      description
+        "References a defined AS path set by name";
+    }
+  }
+
+  grouping set-as-path-reference-state {
+    description
+      "Operational state data for referening an as-path-set
+      in the set-as-path action";
+  }
+
+  grouping set-as-path-reference-top {
+    description
+      "Top-level grouping for referening an as-path-set in the
+      set-as-path action";
+
+    container reference {
+      when "../config/method='REFERENCE'" {
+        description
+          "Active only when the set-as-path method is REFERENCE";
+      }
+      description
+        "Provide a reference to an as-path-set for the
+        set-as-path action";
+
+      container config {
+        description
+          "Configuration data for referening an as-path-set
+          in the set-as-path action";
+
+        uses set-as-path-reference-config;
+      }
+
+      container state {
+
+        config false;
+
+        description
+          "Operational state data for referening an
+          as-path-set in the set-as-path action";
+
+        uses set-as-path-reference-config;
+        uses set-as-path-reference-state;
+      }
+    }
+  }
+
+  grouping set-as-path-action-config {
+    description
+      "Configuration data for the set-as-path action";
+
+    // TODO(xqm): should we rename this to set-bgp-defined-set-action-common?
+    uses set-community-action-common;
```

Comment:
@dplore I am reusing the `INLINE/REFERENCE` method defined in `set-community-action-common`. Ideally we should rename this so that it's not specific to communities. However, it might introduce some breaking change IIUC. Please advise. Thanks!

Reply:
IMO, we should be verbose and create a new grouping for aspath INLINE/REFERENCE.

Reply:
Created `set-as-path-action-config` and new option type `bgp-set-as-path-option-type`.